### PR TITLE
Implemented ScenarioSimulator.WorkerRole

### PR DESCRIPTION
**Pull request by hveiras**
- Implemented ScenarioSimulator.WorkerRole
- Added support for SourceName and InstanceName when running in the context of a WebJob
- Updated Azure SDK references to 2.7
- Introduced support for Roles configuration removed in #255

connects to #84 
connects to #261
connects to #233

### CODE REVIEW

File: src/LongTermStorage/DotnetEventProcessor/ColdStorage.EventProcessor.ConsoleHost/ColdStorage.EventProcessor.ConsoleHost.csproj [hunk 1]
```diff
@@ -36,16 +36,16 @@
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="Microsoft.Data.Edm, Version=5.6.4.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.Data.Edm.5.6.4\lib\net40\Microsoft.Data.Edm.dll</HintPath>
+      <Private>True</Private>
```
Comment (bennage) [inline src/LongTermStorage/DotnetEventProcessor/ColdStorage.EventProcessor.ConsoleHost/ColdStorage.EventProcessor.ConsoleHost.csproj]: What's the implication of `<Private>True</Private> `?

**Comment by bennage** [comment]
On the whole, this looks good. I'd like to know a little bit about the changes to the `<Reference />` elements. Is that something VS did?

**Comment by bennage** [comment]
Can you also add the worker role project to the sln for the Simulator?

File: src/LongTermStorage/DotnetEventProcessor/ColdStorage.EventProcessor.ConsoleHost/ColdStorage.EventProcessor.ConsoleHost.csproj (hunk 1 above)
Comment (hveiras) [inline src/LongTermStorage/DotnetEventProcessor/ColdStorage.EventProcessor.ConsoleHost/ColdStorage.EventProcessor.ConsoleHost.csproj]: This is something that happened after executing Update-Package –reinstall (to re-target a project I initially created as net45 but should be net451

File: src/LongTermStorage/DotnetEventProcessor/ConsumerGroupsDeployment/ServiceConfiguration.Cloud.cscfg [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ServiceConfiguration serviceName="ConsumerGroupsDeployment" xmlns="http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceConfiguration" osFamily="4" osVersion="*" schemaVersion="2015-04.2.6">
+  <Role name="ConsumerGroups.WorkerRole">
+    <Instances count="1" />
+    <ConfigurationSettings>
+      <Setting name="Coldstorage.ConsumerGroupName" value="ColdStorage" />
+      <Setting name="Coldstorage.CheckpointStorageAccount" value="DefaultEndpointsProtocol=https;AccountName=[YourStorageAccount];AccountKey=[YourAccountKey]" />
```
Comment (francischeung) [inline src/LongTermStorage/DotnetEventProcessor/ConsumerGroupsDeployment/ServiceConfiguration.Cloud.cscfg]: Looks like we need to update the setup scripts to fill in these values.

File: src/Simulator/ScenarioSimulator.ConsoleHost/ScenarioSimulator.ConsoleHost.csproj [hunk 3]
```diff
@@ -97,4 +98,5 @@
   <Target Name="BeforeBuild">
     <Error Condition=" '$(CIBuild)' == '' AND !Exists('..\ScenarioSimulator.ConsoleHost.config')" Text="Before building this project, please copy ScenarioSimulator.ConsoleHost.Template.config to ScenarioSimulator.ConsoleHost.config (under the &quot;$([System.IO.Path]::GetFullPath('$(MSBuildProjectDirectory)/../'))&quot; directory) and replace the placeholders in the copy with your configuration information." />
   </Target>
+  <Import Project="..\..\packages\Microsoft.Web.WebJobs.Publish.1.0.3\tools\webjobs.targets" Condition="Exists('..\..\packages\Microsoft.Web.WebJobs.Publish.1.0.3\tools\webjobs.targets')" />
```
Comment (francischeung) [inline src/Simulator/ScenarioSimulator.ConsoleHost/ScenarioSimulator.ConsoleHost.csproj]: I'm a bit confused. Are we promoting BOTH WebJobs and Worker Roles?

Comment (bennage) [inline src/Simulator/ScenarioSimulator.ConsoleHost/ScenarioSimulator.ConsoleHost.csproj]: That wasn't my intent. I wanted to switch to Worker Roles. I did say that removing the  Web Job was secondary to implementing the Worker Role though. I suspect that this was something VS added.

Comment (hveiras) [inline src/Simulator/ScenarioSimulator.ConsoleHost/ScenarioSimulator.ConsoleHost.csproj]: Yes, that package is imported by VS when you publish the WebJob

File: src/LongTermStorage/DotnetEventProcessor/ConsumerGroupsDeployment/ServiceConfiguration.Cloud.cscfg (hunk 2 above)
Comment (hveiras) [inline src/LongTermStorage/DotnetEventProcessor/ConsumerGroupsDeployment/ServiceConfiguration.Cloud.cscfg]: Yes you are right. I added a task in the backlog. It is probably a good idea not to merge this until we have the scripts updated.

**Comment by bennage** [comment]
Sorry, I forgot this was assigned to me. I'll pick it back up today.

**Comment by bennage** [comment]
I merged this after a local rebase.

**Pull request closed without merging**